gimp/pictonode-gimp-plugin/GIMP 2.99/pictonode/custom_nodes.py

```diff
@@ -0,0 +1,75 @@
+# This file was written by Parker Nelms and Stephen Foster.
+
+from httpclient import *
+from client import *
+import window
+import pickle
+
+import sys
+import threading
+import os
+import uuid
+import ontario
+
+# autopep8 off
+import gi
+gi.require_version("GIRepository", "2.0")
+from gi.repository import GIRepository  # noqa
+
+GIRepository.Repository.prepend_search_path(
+    os.path.realpath(
+        os.path.dirname(
+            os.path.abspath(__file__)) +
+        "/introspection"))
+
+GIRepository.Repository.prepend_library_path(
+    os.path.realpath(
+        os.path.dirname(
+            os.path.abspath(__file__)) +
+        "/libs"))
+
+gi.require_version("GtkNodes", "0.1")
+from gi.repository import GtkNodes  # noqa
+
+gi.require_version('Gimp', '3.0')
+from gi.repository import Gimp  # noqa
+
+gi.require_version('GimpUi', '3.0')
+from gi.repository import GimpUi  # noqa
+
+gi.require_version('Gegl', '0.4')
+from gi.repository import Gegl  # noqa
+
+gi.require_version("Gio", "2.0")
+from gi.repository import Gio  # noqa
+
+gi.require_version("GLib", "2.0")
+from gi.repository import GLib  # noqa
+
+gi.require_version("GObject", "2.0")
+from gi.repository import GObject  # noqa
+
+gi.require_version('Gtk', '3.0')
+from gi.repository import Gtk  # noqa
+
+gi.require_version('Gdk', '3.0')
+from gi.repository import Gdk  # noqa
+
+gi.require_version('GdkPixbuf', '2.0')
+from gi.repository import GdkPixbuf  # noqa
+# autopep8 on
+
+# node classes based on examples from img.py in the gtknodes project
+
+# First, define different node types using GtkNodes.Node as a base type
+# implement gegl node operations using ontario backend
+
+class ImgSrcNode(GtkNodes.Node):
+    __gtype_name__ = 'SrcNode'
+
+    def __init__(self, *args, **kwds) -> None:
+        super().__init__(*args, **kwds)
+
+        # build an image contexts 
+        # add nodes to it 
```

Comment:
Did the file get cut off here?

Reply:
oops didnt see this get included in changes